cmd/geth/main.go

```diff
@@ -381,7 +382,9 @@ func run(ctx *cli.Context) {
 	}
 
 	cfg := utils.MakeEthConfig(ClientIdentifier, nodeNameVersion, ctx)
-	cfg.ExtraData = makeDefaultExtra()
+	if len(cfg.ExtraData) == 0 {
+		cfg.ExtraData = makeDefaultExtra()
+	}
```

Comment:
The problem with this is that is makes it impossible to set the extra field to empty. If we want to make this thing configurable, a better approach would be not to have this check here (and not to modify it here at all), rather when processing the flag check if it was set (~ctx.IsSet(ExtraDataFlag.Name)`), and if not, `makeDefaultExtra`, otherwise use whatever was set (even if empty).

That would entail moving the `makeDefaultExtra` into `cmd/utils`, but that should be fine imho.